src/Operation/Watch.php

```diff
@@ -186,6 +187,12 @@ public function __construct(Manager $manager, $databaseName, $collectionName, ar
     /** @internal */
     final public function commandStarted(CommandStartedEvent $event)
     {
+        if ($event->getCommandName() !== 'aggregate') {
+            return;
+        }
+
+        $this->operationTime = null;
```

Comment:
Why do we need to set `operationTime` here?

Reply:
Just being a bit defensive here. If `aggregate` fails, I didn't want any stale values to remain in place. Granted, if it _does_ fail then the change stream will expose the exception and won't bother to resume anyway.

Now that you bring this up, I do think there's a subtle behavioral difference with this PR in that we now _always_ collect `operationTime` from _every_ `aggregate` response. Previously, we would only collect `operationTime` (and use APM) for the very first `aggregate` command executed by a Watch instance, and skip APM if `$this->operationTime` was already set going into a second `aggregate` execution (i.e. resume attempt). [Resume Process](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#resume-process) in the spec does refer to "original aggregation", but I wasn't sure if that was simply to differentiate from a `getMore` command or it was actually specifying _not_ to capture an operation time from a resuming `aggregate`.

I don't see why a resuming `aggregate`'s operation time would be undesirable, but it would probably be best to revert this to the original behavior and only capture `operationTime` on the very first `aggregate`.

Note: this will be a moot point for 4.2+, since `postBatchOperationTime` will be available for empty batches. That said, it's still very relevant for a 4.0 server.